components/script/dom/bindings/structuredclone.rs

```diff
@@ -67,16 +67,49 @@ unsafe fn read_length(r: *mut JSStructuredCloneReader)
   return length as usize;
 }
 
+struct StructuredCloneWriter {
+    w: *mut JSStructuredCloneWriter,
+}
+
+impl StructuturedCloneWriter {
+    unsafe fn write_slice(&self, v: &[u8]) {
+        assert!(JS_WriteBytes(self.w, v.as_ptr() as *const raw::c_void, v.len()));
```

Comment:
so should I call write_length function in CloneWriter Struct and pass that to a variable then use that for length?